### PR TITLE
SS-495 implement test with pytest django

### PR DESCRIPTION
**Pull request by sandstromviktor**
## Status

- [X] Ready
- [ ] Draft
- [ ] Hold

## Description
[JIRA LINK](https://scilifelab.atlassian.net/browse/SS-495?atlOrigin=eyJpIjoiM2IwYzJiYTI0YmNmNGYzMWJkYTNhOTcxN2E2ZGUyMDkiLCJwIjoiaiJ9)

I've integrated `pytest-django` into stackn.
It seems that there are tests in `apps`, `models`, `projects`, and `studio`. Theses tests are `django.test` and based on `unittest`. 
Two other folder contained empty test files, `monitor` and `portal`. I have added two simple tests to `portal`. 

The neat thing is that running `sh -c pytest` runs all tests. 
In `docker-compose.yaml`, i have added a `service` called `tests` and set it to not run when running `up`. The tests are then executed by running `docker-compose run tests` which i think is a neat way of doing it. Agree?
I've also added a workflow step that runs the tests.

To summarize:

- Added `pytest-django` to stackn
- Added 2 tests to `portal`
- Added a test-service to the compose file
- Added a workflow that runs tests inside `build_and_test_studio`. 
- Added coverage to the tests. It prints a large table showing the coverage of every file. 

## Types of changes

What types of changes does your code introduce to STACKn?

- [ ] Hotfix (fixing a critical bug in production)
- [ ] Bugfix
- [X] New feature
- [ ] Documentation update

## Checklist

_If you're unsure about any of the items below, don't hesitate to ask. We're here to help! 
This is simply a reminder of what we are going to look for before merging your code._

- [X] This pull request is against **develop** branch (not applicable for hotfixes)
- [ ] I have included a link to the issue on GitHub or JIRA (if any)
- [ ] I have included migration files (if there are changes to the model classes)
- [X] I have read the [CONTRIBUTING](https://github.com/scaleoutsystems/stackn/blob/master/CONTRIBUTING.md) doc
- [X] I have included tests to complement my changes
- [ ] I have updated the related documentation (if necessary) 
- [ ] I have updated the release notes (docs/releasenotes.md)
- [X] I have added a reviewer for this pull request
- [X] I have added myself as an author for this pull request

### CODE REVIEW

File: portal/tests.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    request.session = s
+
+    # Get response. Since index is a function, this is the correct way
+    response = views.index(request)
+
+    # Check if it returns the correct status code
+    assert response.status_code == 200
+
+
+def test_HomeView():
```
Comment (churnikov) [inline portal/tests.py:27]: I think this is more in line with pep8

```suggestion
def test_home_vew_class():
```

**Review by churnikov — APPROVED** [review]
> The tests are then executed by running docker-compose run tests which i think is a neat way of doing it. Agree?

Wonderful!

**Comment by churnikov** [comment]
But regarding test coverage, may be we could make it separate compose run?

File: portal/tests.py (hunk 1 above)
Comment (sandstromviktor) [reply portal/tests.py]: I agree, but with `view` instead of `vew` hehe

**Review by alfredeen — APPROVED** [review]
Nice work

**Pull request merged**